src/metabase/driver/common/parameters/values.clj

```
@@ -264,8 +268,6 @@
     ;; If the param is not present in `params` use a default from either the tag or the Dashboard parameter.
     ;; If both the tag and Dashboard parameter specify a default value, prefer the default value from the tag.
     (or (:value matching-param)
-        (when (and nil-value? (:required tag))
-          (throw (missing-required-param-exception (:display-name tag))))
```

The only thing that gives me pause is that this was clearly an intentional decision to throw in this case (and above), but the rational has been lost to the mists of time. Feels like "Chesterton's fence". Trolling through the comments in https://github.com/metabase/metabase/pull/31891 I don't see anything related to this specific point. That PR seemed to be more focused on enabling the case of overriding the default for non-required params, so hard to tell if there was any strong reason to make it an error for required params.

The only thing I can think is maybe this is the behavior you want for programatic API calls. Like if the user clears the value in the UI, using the default seems ok, but if someone writes a script that hits the API endpoints, providing a nil value for a required param might be a programming error and you want the thing to fail loudly so you know to go fix your script. Not a very compelling argument though.

---

At a bare minimum, the new behavior was discussed and decided upon [here](https://metaboat.slack.com/archives/C0645JP1W81/p1719233020129649), and various people weighed in on it.